### PR TITLE
WIP: end-to-end experience with macOS package

### PR DESCRIPTION
**Pull request by guillaumerose**
Instead of using embedded files in crc, this use a macOS package to ship all our binaries.

Installer from the CI without the bundle:  https://6736-165193530-gh.circle-artifacts.com/0/crc/macos-amd64/crc-installer.pkg 

Screenshot 2021-01-08 at 14 07 03

After install:
```
$ tree /Library/crc/1.20.0 
/Library/crc/1.20.0
├── CodeReady\ Containers.app
│   └── Contents
│       ├── Frameworks
│       │   ├── libswiftAppKit.dylib
│       │   ├── libswiftCore.dylib
│       │   ├── libswiftCoreData.dylib
│       │   ├── libswiftCoreFoundation.dylib
│       │   ├── libswiftCoreGraphics.dylib
│       │   ├── libswiftCoreImage.dylib
│       │   ├── libswiftDarwin.dylib
│       │   ├── libswiftDispatch.dylib
│       │   ├── libswiftFoundation.dylib
│       │   ├── libswiftIOKit.dylib
│       │   ├── libswiftMetal.dylib
│       │   ├── libswiftObjectiveC.dylib
│       │   ├── libswiftQuartzCore.dylib
│       │   ├── libswiftXPC.dylib
│       │   └── libswiftos.dylib
│       ├── Info.plist
│       ├── MacOS
│       │   └── CodeReady\ Containers
│       ├── PkgInfo
│       ├── Resources
│       │   ├── AppIcon-1.icns
│       │   ├── Assets.car
│       │   ├── Base.lproj
│       │   │   └── Main.storyboardc
│       │   │       ├── Aed-9x-vsJ-view-voe-Te-Ymr.nib
│       │   │       ├── Info.plist
│       │   │       ├── MainMenu.nib
│       │   │       ├── NSWindowController-I4M-tg-MG9.nib
│       │   │       ├── NSWindowController-OEv-Wj-D9q.nib
│       │   │       ├── NSWindowController-SbK-ai-iAt.nib
│       │   │       ├── V0P-Gq-7lO-view-Chj-ud-as9.nib
│       │   │       ├── WBr-05-0in-view-ZEr-oV-ns0.nib
│       │   │       └── nSR-KS-iPn-view-gzA-fr-McP.nib
│       │   ├── ExportOptions.plist
│       │   └── pullSecretStoryBoard.storyboardc
│       │       ├── 0oN-MP-yQV-view-ZRY-7T-CBv.nib
│       │       ├── Info.plist
│       │       └── PullSecretWindowController.nib
│       └── _CodeSignature
│           └── CodeResources
├── admin-helper
├── crc
├── crc-driver-hyperkit
├── crc_hyperkit_4.6.6
│   ├── crc-bundle-info.json
│   ├── crc.qcow2
│   ├── id_rsa_crc
│   ├── initramfs-4.18.0-193.29.1.el8_2.x86_64.img
│   ├── kubeadmin-password
│   ├── kubeconfig
│   ├── oc
│   └── vmlinuz-4.18.0-193.29.1.el8_2.x86_64
├── crc_hyperkit_4.6.6.crcbundle
└── hyperkit

0 directories, 6 files
```

### CODE REVIEW

File: packaging/darwin/Distribution [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8" standalone="no"?>
+<installer-script minSpecVersion="1.000000">
+    <title>CodeReady Containers</title>
+    <background mime-type="image/png" file="banner.png" scaling="proportional"/>
```
Comment (anjannath) [inline packaging/darwin/Distribution:4]: need to also add for the banner to appear in systems using dark-mode, ref: https://developer.apple.com/forums/thread/108056
```xml
<background-darkAqua mime-type="image/png" file="banner.png" scaling="proportional"/>
```

File: cmd/crc/cmd/delete.go [hunk 2]
```diff
@@ -15,8 +15,10 @@ import (
 var clearCache bool
 
 func init() {
-	deleteCmd.Flags().BoolVarP(&clearCache, "clear-cache", "", false,
-		fmt.Sprintf("Clear the OpenShift cluster cache at: %s", constants.MachineCacheDir))
+	if os.Getenv("CRC_PACKAGE") == "" {
```
Comment (gbraad) [inline cmd/crc/cmd/delete.go:18]: cache lives in `~/.crc/cache`, right? or is this also modified

**Review by gbraad — CHANGES_REQUESTED** [review]
looks a little hacky, any plans to refine this?

I am a little concerned about the removal of the `setup` and `cleanup`. as it means now that a `start` could detect issues, but has no way to resolve them without reinstallng, right?

File: cmd/crc/cmd/delete.go (hunk 2 above)
Comment (praveenkumar) [reply cmd/crc/cmd/delete.go:18]: No it is changed https://github.com/code-ready/crc/pull/1827/files#diff-351617b3a8d07ea69e3c3430ca45061b17effc84aba09b482e8b8d79ee696a4eR110-R117 here

**Comment by guillaumerose** [comment]
`crc setup` and `crc cleanup` do very little on macOS. It is just few lines of shell. 

I will remove the CRC_PACKAGE env var. and do something at compile time. 
I will also try to have the same structure in /Library/crc/1.20.0 as ~/.crc and adds bin/ and cache/. 

To remove one hack, I need https://github.com/code-ready/admin-helper/pull/5 :)

**Comment by anjannath** [comment]
> crc setup and crc cleanup do very little on macOS. It is just few lines of shell.

Putting it here so that we don't miss it later, It also creates (and remove) the launchd files for the tray and daemon

**Comment by guillaumerose** [comment]
Will re-open smaller PRs. Keeping this one in archive :)

**Pull request closed without merging**